dev_tools/utils/nxdl_utils.py

```diff
@@ -783,17 +759,15 @@ def walk_elist(elist, html_name):
                     if fitting_child is not None:
                         child = fitting_child
                     break
-        elist[ind] = child
-        if elist[ind] is None:
-            del elist[ind]
+        if child is None:
             continue
         # override: remove low priority inheritance classes if class_type is overriden
         if len(elist) > ind + 1 and get_nx_class(elist[ind]) != get_nx_class(
             elist[ind + 1]
         ):
             del elist[ind + 1 :]
         # add new base class(es) if new element brings such (and not a primitive type)
-        if len(elist) == ind + 1 and get_nx_class(elist[ind])[0:3] != "NX_":
+        if len(elist) == ind + 1 and get_nx_class(elist[ind]).startswith("NX_"):
```

Comment:
functionally different startswith("NX_") will eval to True if left-hand side variable starts indeed with "NX_" but on the LHS the statement checks that the left-hand side variable DOES NOT START WITH "NX_" !

Reply:
The comment above states the intended functionality, I believe. It wants to check `not a primitive type`.
I add the `not`.

```suggestion
        if len(elist) == ind + 1 and not get_nx_class(elist[ind]).startswith("NX_"):
```